File: util/src/main/java/net/yakclient/graphics/util/YakTexture.java
package net.yakclient.graphics.util;

public interface YakTexture {
    boolean hasAlpha();

    void bind();

    int getImageHeight();

    int getImageWidth();

    float getHeight();

    float getWidth();

    int getTextureHeight();

    int getTextureWidth();

    void release();

    TexAggregation generateTexs(VerticeAggregation vertices);

}

File: util/src/main/kotlin/net/yakclient/graphics/util/YakTexture.kt
package net.yakclient.graphics.util

public interface YakTexture {
    public val parent: YakTexture?

    public val height: Int
    public val width: Int
    public val offsetX: Int
    public val offsetY: Int

    public fun subTexture(x: Int, y: Int, width: Int, height: Int): YakTexture

    public fun bind()
    public fun release()
}